### DISCUSSION TITLE
Can the error severity be removed from the error message created by Postgresql

### DISCUSSION THREAD

**Discussion by AbhinashJha342** [docker-library/postgres#1268] [General]
Is there a way to remove the "ERROR:" from the error message that postgresql sends? Adding a sample error message :

```
ERROR: insert or update on table "bonus" violates foreign key constraint "bonus_cust_id_fkey"
  Detail: Key (cust_id)=(1998) is not present in table "customer"
```

All SQLException thrown by Postgres seems to be starting with "ERROR:" and I am exploring options if it can be removed through superuser.

I have raised a similar question [here](https://superuser.com/questions/1853088/remove-the-severity-of-the-error-from-error-message-send-by-postgresql).

Any suggestions or confirmation if it can or cannot be done is appreciated.
Thanks!

**Comment by tianon** [comment]
The best I can find is https://www.postgresql.org/docs/current/runtime-config-logging.html#GUC-LOG-MIN-ERROR-STATEMENT -- ie, setting `log_min_error_statement=PANIC`, which will remove the `ERROR` logs entirely (which doesn't sound like it's what you want, unfortunately).

There does appear to be a JSON format for the logs (https://www.postgresql.org/docs/current/runtime-config-logging.html#RUNTIME-CONFIG-LOGGING-JSONLOG) which might be interesting and allow you to better parse/filter these messages with another tool?

**Reply by AbhinashJha342** [reply]
actually, its a big application with many sql messages, and removing "ERROR:" keyword from the message after SQLException is caught is the last resort. on debugging, I find that the `toString()` method of `ServerErrorMessage` class creates the final message. I was just thinking if there is a way to intercept this method call and make some changes. I am using Java and Jdbc. Below  is the method from `org.postgresql.util.PSQLException` which calls the `toString()`

```
public PSQLException(ServerErrorMessage serverError, boolean detail) {
        super(detail ? serverError.toString() : serverError.getNonSensitiveErrorMessage(), serverError.getSQLState());
        this.serverError = serverError;
    }
```
